internal/file/file.go

```diff
@@ -0,0 +1,74 @@
+// Package file handles file operations such as read & write.
+package file
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+const filename = "tasks.json"
+
+// File - file generic struct. T generic.
+type File[T any] struct{}
+
+// New instantiates a File struct.
+func New[T any]() *File[T] {
+	return &File[T]{}
+}
+
+// Write - writes tasks into a JSON file
+func (f *File[T]) Write(data *[]T) error {
+	bytes, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("marshalling file: %w", err)
+	}
+
+	err = os.WriteFile(filename, bytes, 0644)
+	if err != nil {
+		return fmt.Errorf("writing file: %w", err)
+	}
+
+	return nil
+}
+
+// Read - reads tasks from json file
+func (f *File[T]) Read() ([]T, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, nil
+		}
+
+		return nil, fmt.Errorf("opening file: %w", err)
+	}
+
+	defer closer(file, &err, "close file")
+
+	bytes, err := os.ReadFile(file.Name())
```

Comment:
Everything above this line can be removed. Except that check, it needs to be done just after ReadFile
```
if errors.Is(err, os.ErrNotExist) {
	return nil, nil
}
```

Reply:
Actually, you know what? if you have any doubts or curiosity (and you should be curious), you can easily just click on the function name in the docs and get to its source code. And you should read many things like that.
Here is ReadFile, for example.

https://pkg.go.dev/os#ReadFile